de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/InstantIncomingProjectNegotiation.java

```diff
@@ -64,30 +62,6 @@ protected void transfer(
       IProgressMonitor monitor, Map<String, IProject> projectMapping, List<FileList> missingFiles)
```

Comment:
moving logic of the transfer method to the abstract class interferes with Instant Session Start as it will slightly differ with #156

Reply:
In that case rather then moving the queueing logic out of the `transfer` method into the `AbstractIncomingProjectNegotiation` we should move it into a new method (`handleQueueing` or similar). The whole reason I have moved the code is [this `isHost` check](https://github.com/saros-project/saros/pull/355/files#diff-ba48fd2931f14e2e2d97b270df6eac2eR169), which then could still be handled by the `AbstractIncomingProjectNegotiation` (after the `handleQueueing` call and before the `transfer` call).

Right now everything is neatly implemented on the `AbstractIncomingProjectNegotiation` layer and I would like to avoid having to move parts of this patch into the actual implementations.

Do you think that could work out?